src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -4,22 +4,43 @@
 using System;
 using System.Collections.Generic;
 using System.Composition;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Client;
 using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
 {
     [Shared]
     [ExportLspMethod(Methods.TextDocumentReferencesName)]
-    internal class FindAllReferencesHandler : IRequestHandler<ReferenceParams, VSReferenceItem[]>
+    internal class FindAllReferencesHandler : IRequestHandler<ReferenceParams, VSReferenceItem[]>, IDisposable
     {
         private readonly LSPRequestInvoker _requestInvoker;
         private readonly LSPDocumentManager _documentManager;
         private readonly LSPProjectionProvider _projectionProvider;
         private readonly LSPDocumentMappingProvider _documentMappingProvider;
 
+        private readonly TimeSpan WaitForProgressCompletionTimeout = new TimeSpan(0, 0, 15);
+
+        // Roslyn sends Progress Notifications every 0.5s *only* if results have been found.
+        // Consequently, at ~ time > 0.5s ~ after the last notification, we don't know whether Roslyn is
+        // done searching for results, or just hasn't found any additional results yet.
+        // 
+        // https://github.com/dotnet/roslyn/blob/1462b1e9c78a7efe338a6859ee5f88fa07d4adc8/src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs#L52-L55
```

Comment:
This link seems out of place for this comment (since it links to the async work queue in roslyn). I'd either remove it or add more context as to why you're including it here